riscv/csrs.cc

```diff
@@ -996,7 +996,19 @@ reg_t wide_counter_csr_t::read() const noexcept {
 }
 
 void wide_counter_csr_t::bump(const reg_t howmuch) noexcept {
-  val += howmuch;  // to keep log reasonable size, don't log every bump
+  auto config_val = 0;
+
+  if (proc->extension_enabled_const(EXT_SMCNTRPMF)) {
+    if (address == CSR_MCYCLE) {
+      config_val = state->csrmap[CSR_MCYCLECFG].get()->read();
```

Comment:
I think you can just add mcyclecfg and following minstretcfg to the state structure, and use them directly.

Reply:
Yes, that is preferable to using `csrmap` which requires a map lookup every time. Though in this case, adding these to the state structure is not necessary if the config register is passed in to the constructor, as I requested in another comment.